src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -193,6 +229,15 @@ public User updateUser(User user) {
     return userToBeUpdated;
   }
 
+  public void updatePreferences(User user){
+      //Getting the correct user (404 and 409 should be check by specific access already)
+      User userToUpdatePreferences = getUserById(user.getId());
+      //Update User Preferences
+      userToUpdatePreferences.setMinAge(user.getMinAge());
+      userToUpdatePreferences.setMaxAge(user.getMaxAge());
+      userToUpdatePreferences.setGenderPreferences(user.getGenderPreferences());
+  }
+
```

Comment:
I would propose to make one big updateUser method, and only update the values that change. I think for readability you can do something like:
`<condition> ? userToBeUpdated.updateField(user.getField()) : null`
, but better check if that really works.

Reply:
I think i would keep the two different methods, as you update the user himself once and once his preferences. Its also easier to read this way and easier to modify

Reply:
Also, what happens if the client does not pass a genderPreference, i.e. does not update it? It would get assigned null? I would add some second lifeline here, so that the server also checks if the values passed are legit (I mean, if we can trick the system via Postman, everyone can). (check with something like userTUP.getMaxAge().equals(null) or smth.)

Reply:
I don't save and flush (actually we also don't on "update user". Should we?

Reply:
For min and max age you can't pass null, as they are ints